src/cache.rs

```diff
@@ -1,4 +1,9 @@
 // This Source Code Form is subject to the terms of the Mozilla Public
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
+use std::fmt::Debug;
 
+pub trait Cache: Debug {
+    fn copy(&self) -> Box<Cache>;
+    fn get_id(&self) -> String;
```

Comment:
I think we've got work to do in this area, so this is not intended to be a final step.  I'm just trying to replicate what we had in the C version.

In this PR, I'm using it to return an "ID" to the dbus calls ListDevs and ListCacheDevs.  This will be displayed to the user in the CLI.  Similar to how ListPools uses the key of the map.

Does this make sense?


In my next PR, I need a way to uniquely identify a element in the Vec to so I can delete it.